.github/workflows/test.yml

```
@@ -9,11 +9,11 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node: [16, 18]
+        node: [20]
```

```suggestion
        node: [18, 20]
```

we may still want to support Node 18?

---

> we may still want to support Node 18?

@tuyennhv It looks like in Lodestar we drop all support for Node 18. In particular, Lodestar workflows are all using node20 according to chainsafe/Lodestar#5730. Do we want workflows in ssz to be consistent with Lodestar's?

---

as a library I think we should support NodeJS 18 until it's end of life (which is Apr 30 2025)
it could be consumed by different applications, not just lodestar
cc @matthewkeil @nflaig

---

For Lodestar, as in the client, we wanna support the node versions we actually test against, since there is no fleet running v18, it's safer to drop support than have users run into unexpected issues that are hard to debug.

For libraries, we should try to support node versions that are still in active maintenance, including 18 right now.

My idea was that once we start using node 22 that even in the monorepo we still run unit tests against both 20/22. The only tradeoff is that it adds additional load to CI but should be fine for unit.

---

> It looks like in Lodestar we drop all support for Node 18. 

This also doesn't drop support for node 16, unless we added code changes that rely on another minimum node version (which should be communicated in the release notes)

---

> as a library I think we should support NodeJS 18 until it's end of life (which is Apr 30 2025) it could be consumed by different applications, not just lodestar cc @matthewkeil @nflaig

Agreed.  We should definitely support 18 for this lib, and when 22 is ready we should add but not drop 18 until 18 has reached end of life